js/msp.js

```diff
@@ -965,8 +963,28 @@ var MSP = {
                 break;
 
             case MSP_codes.MSP_PID_ADVANCED:
-                if (semver.gte(CONFIG.apiVersion, "1.16.0")) {
-                    var offset = 0;
+                var offset = 0;
+                if (semver.gte(CONFIG.flightControllerVersion, "3.0.0")) {
```

Comment:
Agree with you on just about all points, except that there is no reason whatsoever to replace my correct checks for apiVersion with checks for flightControllerVersion, particularly if apiVersion is used everywhere else in this file.